api/client.go

```diff
@@ -309,6 +311,9 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 	if err := c.validateCharmVersion(ch); err != nil {
 		return nil, errors.Trace(err)
 	}
+	if err := validateCharmLXDProfile(ch); err != nil {
```

Comment:
Perhaps a comment on why we wait to this point to validate the charm.  Per conversation.